### ISSUE TITLE
Can't connect to cache.nixos.org nor nix-community.cachix.org

### ISSUE THREAD

**Issue by rgnns** [NixOS/nix#5332]
**Describe the bug**

After installing Nix with a multi-user installation on macOS, can't build nor install because Nix is unable to create any connection. Running a command such as `niv-env -i direnv` (or any other package) will result in 

```
warning: ignoring untrusted substituter 'https://cache.nixos.org'
warning: ignoring untrusted substituter 'https://nix-community.cachix.org'
warning: ignoring the user-specified setting 'trusted-public-keys', because it is a restricted setting and you are not a trusted user
...
warning: unable to download 'http://tarballs.nixos.org/sha256/6fae5d5db226ea8eace94f2d0221a947a48c7ef7e7fd515c6604c705eb49662b': Couldn't resolve host name (6); retrying in 250 ms
warning: unable to download...
```

**Steps To Reproduce**

1. Install Nix on mac using `curl -L https://nixos.org/nix/install | sh -s -- --darwin-use-unencrypted-nix-store-volume --daemon`
1. Restart device.
1. Try to install a packages: `nix-env -i <package>`.
1. See error.

**Expected behavior**

Should be able to install packages and run `nix-build` successfully.

**`nix-env --version` output**

nix-env (Nix) 2.3.15

**Additional context**

***OS**: macOS Big Sur 11.6, kernel version 20.6.0, x86_64</note>*
I've successfully installed and used Nix on other macOS computers, but it is not working on this one for some reason. I couldn't find troubleshooting guides to problems like this and I'm not sure what might be the reason, I thought it might be a problem with proxies (a similar issue happen when I use `bazel`) but I can't find how to resolve this for Nix.

**Comment by abathur** [comment]
(Never tried to use Nix through a proxy so I may be clueless here; just taking a few stabs at collecting detail...)

> I thought it might be a problem with proxies (a similar issue happen when I use bazel)

Can you clarify: 
- you _are_ using a proxy on this system?
- you _were not_ using a proxy with the other devices you installed Nix on?
- does https://tarballs.nixos.org load fine in a browser on the system? (it looks like cache.nixos.org and nix-community.cachix.org are just warnings, but the tarball download fails with what sounds like a DNS error)
- does `curl https://tarballs.nixos.org` print the html for the same page as above?

**Comment by rgnns** [comment]
@abathur this is a managed computer, so I'm not doing anything that I'm aware of. Looking at network proxies it seems to be disabled, but perhaps it has been set in some other way I'm unaware of. All system environment variables for proxy setting doesn't seem to be set. I'm sure that for all other devices in which I installed Nix have no proxies. https://tarballs.nixos.org does load from a browser and running `curl` on it also prints the html. I have added the cache.nixos.org and nix-community.cachix.org warnings because they seem to be a hint as to what is going on (I don't get these warnings on my other devices).
